expression/builtin_control.go

```diff
@@ -338,7 +338,7 @@ func (c *ifFunctionClass) getFunction(args []Expression, ctx context.Context) (s
 	return sig.setSelf(sig), nil
 }
 
-func (c *ifFunctionClass) inferType(tp1, tp2 *types.FieldType) *types.FieldType {
+func inferType(tp1, tp2 *types.FieldType) *types.FieldType {
```

Comment:
add a comment for this, better refine the function name, if it's only used by if, ifnull.